### PR TITLE
RFC, IANAL: clarify some license special cases

### PR DESCRIPTION
**Pull request by tkelman**
FFTW wrappers were moved and mostly rewritten since the line about base/fftw.jl in LICENSE.md was added - ref 481f351 and #8248 and #12201 - @stevengj what are your thoughts on whether the ccall wrapper functions would be considered a derived work, do you think the the Julia wrappers are GPL or not?

Also add mentions of base/grisu and src/support/strtod.c to the top-level license file.

### CODE REVIEW

**Review by StefanKarpinski — APPROVED** [review]

**Comment by stevengj** [comment]
@tkelman, I don't think the wrapper functions by *themselves* would constitute a derived work (but IANAL, obviously). In any casem even if they were a derived work they would presumably be derived from the `fftw3.h` header file, and [that header is 2-clause BSD](https://github.com/FFTW/fftw3/blob/cc5fc8ce7ffd77f467740554f649aab4d3f71344/api/fftw3.h#L1-L30).

The *combination* of the wrappers (and any code calling the wrappers) and the FFTW library, i.e. software actually *using* FFTW via the wrappers, would be a derived work and subject to the GPL, of course.

**Comment by tkelman** [comment]
Thanks! Useful that the header is BSD-2, was not aware of that.

What I'm partially working towards here is getting `USE_GPL_LIBS = 0` (which disables the building of libfftw, and currently also disables the inclusion of the ccall wrappers into Base) to work correctly together with `USE_INTEL_MKL_FFT = 1`. If we're linking against a differently-licensed implementation of the API, the Julia ccall wrappers will need to be included.

**Comment by stevengj** [comment]
FYI, the `fftw3.h` header was made BSD precisely to allow Intel to (partly) copy our API without worrying about GPL infringement.

**Comment by tkelman** [comment]
(since Jeff added that line)

File: LICENSE.md [hunk 1]
```diff
@@ -70,9 +71,9 @@ their own licenses:
 
 The following components of Julia's standard library have separate licenses:
 
-- base/fftw.jl (see [FFTW](http://fftw.org/doc/License-and-Copyright.html))
```
Comment (tkelman) [inline LICENSE.md]: ping @JeffBezanson thoughts on this?

Comment (StefanKarpinski) [inline LICENSE.md]: The file itself is not and never has been GPL – it's MIT like the rest of Julia. So: ✔️

Comment (stevengj) [inline LICENSE.md]: Yes, but it should still say something about the FFTW dependency if you call the fft etc. routines

Comment (tkelman) [inline LICENSE.md]: probably covered by the "Julia's standard library uses the following external libraries" bullet above, same as for the rest of the ccall dependencies

Comment (tkelman) [inline LICENSE.md]: I just want positive confirmation from Jeff on this, rather than implicit silence, since he originally wrote this line.

Comment (JeffBezanson) [inline LICENSE.md]: Yes this is ok.

**Review by JeffBezanson — APPROVED** [review]

**Pull request merged**